backend/routes/retool.js

```diff
@@ -0,0 +1,33 @@
+var express = require('express');
+var router = express.Router();
+
+  /**
+   * {
+   *  orgId: "1",
+   *  pageUuid: "063ae5a6-5b07-11ed-94b5-f7ac74a32e3a",
+   *  externalIdentifier: "maxantony",
+   *  groupIds: [5,6]
+   * }
+   */
+  router.post('/embedUrl', (req, res) => {
+    const options = {
+      method: "POST",
+      headers: {
+        'Authorization': process.env.RETOOL_API_KEY,
+        'content-type': 'application/json'
+      },
+      body: JSON.stringify({
+        "orgId": "1",
+        "pageUuid": req.body.pageUuid,
+        "externalIdentifier": req.body.externalIdentifier,
+        "groupIds": req.body.groups
```

Comment:
This will work, but to make this more realistic, these parameters should be determined based on the identity of the user logged in to the customer portal (NOT controlled by the frontend). The idea is that the customer's backend is the source of authority for who the user is and what part of Retool can access. If the frontend can determine that, it's insecure.

Reply:
Yeah, totally. Annoying thing is that this codebase uses an Auth0 SDK that determines and maintains the user identity. This actually is the only used route on the backend, so not sure what the best alternative is...